docker/drone/src/main/java/org/arquillian/cube/docker/graphene/location/DockerCubeCustomizableURLResourceProvider.java

```diff
@@ -0,0 +1,335 @@
+        return seleniumContainers.getSeleniumContainerName().equals(cubeContainerEntry.getKey())
+                || seleniumContainers.getVncContainerName().equals(cubeContainerEntry.getKey());
+    }
+
+    private boolean noPreviousBindPortFound(int bindPort) {
+        return bindPort == -1;
+    }
+
+    private boolean hasOnlyOneBindPort(CubeContainer cubeContainer) {
+
+        final Collection<PortBinding> portBindings = cubeContainer.getPortBindings();
+        if (portBindings == null) {
+            return false;
+        }
+
+        return portBindings.size() == 1;
+    }
+
+    private boolean hasMoreThanOneBindPort(CubeContainer cubeContainer) {
+
+        final Collection<PortBinding> portBindings = cubeContainer.getPortBindings();
+        if (portBindings == null) {
+            return false;
+        }
+
+        return portBindings.size() > 1;
+    }
+
+    private int extractPort(String url) {
+        int colonLocation = url.indexOf(':');
+
+        if (colonLocation > 0) {
+            for (int i = colonLocation +1; i < url.length(); i++) {
+                if (url.charAt(i) == '/') {
+                    return Integer.parseInt(url.substring(colonLocation + 1, i));
+                }
+            }
+            return Integer.parseInt(url.substring(colonLocation + 1));
+        }
+
+        return NO_PORT;
+    }
+
+    private String resolveHost(String url) {
+        for (int i=0; i < url.length(); i++) {
+            if (url.charAt(i) == ':' || url.charAt(i) == '/') {
+                return url.substring(0, i);
+            }
+        }
+
+        return url;
+    }
+
+
+    private boolean startWithDockerHost(String url) {
+        return url.startsWith("dockerHost");
+    }
+
+    private boolean isAnAbsoluteUrl(String url) {
+        return url.startsWith("/");
+    }
+
+    private static final class UrlBuilder {
+
+        private String protocol = "http";
+        private String host;
+        private int port = -1;
+        private String file = "";
+
+        private UrlBuilder() {
+            super();
+        }
+
+        public static UrlBuilder create() {
+            return new UrlBuilder();
+        }
+
+        public UrlBuilder protocol(String protocol) {
+            this.protocol = protocol;
+            return this;
+        }
+
+        public UrlBuilder host(String host) {
+            this.host = host;
+            return this;
+        }
+
+        public UrlBuilder port(int port) {
+            this.port = port;
+            return this;
+        }
+
+        public UrlBuilder context(String context) {
+            this.file = context;
+            return this;
+        }
+
+        public URL build() throws MalformedURLException {
+
+            if (protocol == null || protocol.isEmpty()) {
+                throw new IllegalArgumentException("Protocol cannot be null or empty");
+            }
+
+            if (host == null || host.isEmpty()) {
```

Comment:
`String.isEmpty()` only in JDK6 and above :\ I assume we don't support JDK5 anymore as probably other libraries we use are not.